src/yea/cli.py

```diff
@@ -2,29 +2,54 @@
 
 import argparse
 import sys
+from typing import Callable, List, Optional
 
 from yea import context, runner
 
 
-def cli_list(yc):
+if sys.version_info >= (3, 8):
+    from typing import Literal
+else:
+    from typing_extensions import Literal
+
+
+class CliArgs:
```

Comment:
if it's only data without functions, maybe it can be `namedtuple` (or use `dataclass`)?

Reply:
I'd love to use `dataclass`, but didn't want to use a backport to 3.6. @raubitsj, can we drop 3.6 altogether since it's at the end of its lifetime and we're anyways only running functional tests on 3.7 in the client?

Reply:
I'd leave it as is and then when we finally drop 3.6 replace it with a dataclass. (I don't like named tuples :) )

Reply:
just a suggestion... i don't have strong feeling either way :)

Reply:
We should keep py3.6 for now... but you can use this if you want:
https://pypi.org/project/dataclasses/
that is a nice balance of supporting py3.6 but not letting it hold us back.

Reply:
i am also fine with old school class as you did here if you dont want to add a temporary dependency..   The risk is, if yea is depending on dataclasses, and the thing we are testing uses it, we wont know.

(this is happening already as one of the other client dependencies - not sure if it is dev - pulls in dataclasses)

So this is a slight risk, but it is worth it if it makes the code more maintainable.

Reply:
i'd keep the old school way :)